perf/README.md

```diff
@@ -79,11 +79,33 @@ Given you have provisioned your infrastructure, you can now build and run the li
             - `--download-bytes` number of bytes to download per stream.
           - Output
             - Logging MUST go to `stderr`.
-            - Measurement output is printed to `stdout` as JSON in the form of:
-              ```json
-              {"latency": 0.246442851}
-              ```
-              Note that the measurement includes the time to (1) establish the
-              connection, (2) upload the bytes and (3) download the bytes.
+            - Measurement output is printed to `stdout` as JSON.
+            - The output schema is:
+               ``` typescript
+               interface Data {
+                 type: "intermediary" | "final";
+                 timeSeconds: number;
+                 uploadBytes: number;
+                 downloadBytes: number;
+               }
+               ```
+            - Every second the client must print the current progress to stdout. See example below. Note the `type: "intermediary"`.
+               ``` json
+               {
+                 "type": "intermediary",
+                 "timeSeconds": 1.004957645,
+                 "uploadBytes": 73039872,
+                 "downloadBytes": 0
+               },
+               ```
+            - Before terminating the client must print a final summary. See example below. Note the `type: "final"`. Also note that the measurement includes the time to (1) establish the connection, (2) upload the bytes and (3) download the bytes.
+               ``` json
+               {
+                 "type": "final",
+                 "timeSeconds": 60.127230659,
+                 "uploadBytes": 4382392320,
+                 "downloadBytes": 0
+               }
+               ```
```

Comment:
This pull request requires the following changes to a perf binaries output format.

This pull request already updates the `go-libp2p` and `https` implementation. https://github.com/quic-go/perf/pull/17 updates the `quic-go` implementation. https://github.com/libp2p/rust-libp2p/pull/4382 updates the rust-libp2p implementation.

The motivation for the larger effort is documented in https://github.com/libp2p/test-plans/issues/261.

@maschad do you have time and interest to do the corresponding changes on the js-libp2p side? If so, as a first step, you could update the existing `stdout` output to the above detailed `type: "final"` output. As a second step the `type: "intermediate"` can follow.

Reply:
Thanks for the work on this PR @mxinden and the clear write up. I created an issue https://github.com/libp2p/js-libp2p/issues/2064 to track this and will tackle it once I have completed my current in-flight work. Once that is merged and released I will open a PR to https://github.com/libp2p/test-plans